quail/processes/wps_climdex_mmdmt.py

```diff
@@ -0,0 +1,142 @@
+                max_occurs=1,
+                data_type="string",
+            ),
+            LiteralInput(
+                "freq",
+                "Frequency",
+                abstract="Time frequency to aggregate to",
+                allowed_values=["monthly", "annual"],
+                default="monthly",
+                min_occurs=0,
+                max_occurs=1,
+                data_type="string",
+            ),
+            vector_name,
+            log_level,
+        ]
+
+        outputs = [rda_output]
+
+        super(ClimdexMMDMT, self).__init__(
+            self._handler,
+            identifier="climdex_mmdmt",
+            title="Climdex MMDMT",
+            abstract="Monthly or Annual Maximum of Daily Maximum Temperature",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def _handler(self, request, response):
+        climdex_input, ci_name, output_file, month_type, freq, vector_name, loglevel = [
+            arg[0] for arg in collect_args(request, self.workdir).values()
+        ]
+
+        log_handler(
+            self,
+            response,
+            "Starting Process",
+            logger,
+            log_level=loglevel,
+            process_step="start",
+        )
+        climdex = get_package("climdex.pcic")
+
+        log_handler(
+            self,
+            response,
+            "Loading climdexInput from R data file",
+            logger,
+            log_level=loglevel,
+            process_step="load_rdata",
+        )
+        ci = load_rdata_to_python(climdex_input, ci_name)
+
+        log_handler(
+            self,
+            response,
+            f"Processing {month_type}",
+            logger,
+            log_level=loglevel,
+            process_step="process",
+        )
+
+        temps = robjects.r(f"climdex.{month_type}(ci, freq='{freq}')")
+
+        log_handler(
+            self,
+            response,
+            f"Saving {month_type} vector to R data file",
+            logger,
+            log_level=loglevel,
+            process_step="save_rdata",
+        )
+        output_path = os.path.join(self.workdir, output_file)
+        save_python_to_rdata(vector_name, temps, output_path)
+
+        log_handler(
+            self,
+            response,
+            "Building final output",
+            logger,
+            log_level=loglevel,
+            process_step="build_output",
+        )
+        response.outputs["rda_output"].file = output_path
+
+        # Clear R global env
+        robjects.r("rm(list=ls())")
```

Comment:
I don't know why this step is required, would you be able to fill me in?

Reply:
The R env keeps R variables between process runs. `ls()` prints all variables in the R env. After running the processes 5 separate times this is all the variables in the env
```
[1] "another_vector"     "ci"                 "txx_annual_vector" 
[4] "txx_monthly_vector" "vector_3"           "vector_4" 
```
Once the process is completed, we don't need to keep old variables